sklearn/metrics/_classification.py

```diff
@@ -595,7 +595,9 @@ def confusion_matrix(
             UserWarning,
         )
 
-    return xp.asarray(cm, device=device_)
+    return xp.asarray(
+        cm, dtype=_max_precision_float_dtype(xp, device=device_), device=device_
+    )
```

Comment:
This is an unrelated change to this PR but it fails when I run the tests locally so I thought about just fixing it. If required a dedicated PR can be created.